src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -70,42 +106,134 @@ export class HuiCardPicker extends LitElement {
+          cursor: pointer;
+          box-sizing: border-box;
+        }
+
+        .card-header {
+          color: var(--ha-card-header-color, --primary-text-color);
+          font-family: var(--ha-card-header-font-family, inherit);
+          font-size: 16px;
+          letter-spacing: -0.012em;
+          line-height: 20px;
+          padding: 12px 16px;
+          display: block;
+          text-align: center;
+          background: var(
+            --ha-card-background,
+            var(--paper-card-background-color, white)
+          );
+          border-radius: 0 0 4px 4px;
+          border-top: 1px solid var(--divider-color);
+        }
+
+        .preview {
+          pointer-events: none;
+          margin: 20px;
+          flex: 1 1 0;
           display: flex;
-          flex-wrap: wrap;
-          margin-bottom: 10px;
+          align-items: center;
+          justify-content: center;
         }
-        .cards-container mwc-button {
-          flex: 1 0 25%;
-          margin: 4px;
+
+        .preview > :first-child {
+          zoom: 0.6;
+          -moz-transform: scale(0.6); /* Firefox */
+          -moz-transform-origin: 0 0;
+          -o-transform: scale(0.6); /* Opera */
+          -o-transform-origin: 0 0;
+          display: block;
+          width: 100%;
         }
 
-        @media all and (max-width: 450px), all and (max-height: 500px) {
-          .cards-container mwc-button {
-            flex: 1 0 33%;
-          }
+        .description {
+          text-align: center;
+        }
+
+        .spinner {
+          align-items: center;
+          justify-content: center;
         }
       `,
     ];
   }
 
-  private _manualPicked(): void {
-    fireEvent(this, "config-changed", {
-      config: { type: "" },
-    });
+  private _cardPicked(ev: Event): void {
+    const config: LovelaceCardConfig = (ev.currentTarget! as CardPickTarget)
+      .config;
+
+    fireEvent(this, "config-changed", { config });
+  }
+
+  private _createCardElement(cardConfig: LovelaceCardConfig) {
+    const element = createCardElement(cardConfig) as LovelaceCard;
+    element.hass = this.hass;
+    element.addEventListener(
+      "ll-rebuild",
+      (ev) => {
+        ev.stopPropagation();
+        element.parentElement!.replaceChild(
+          this._createCardElement(cardConfig),
+          element
+        );
+      },
+      { once: true }
+    );
+    return element;
   }
 
-  private async _cardPicked(ev: Event): Promise<void> {
-    const type = (ev.currentTarget! as CardPickTarget).type;
+  private async _renderCardElement(
+    type: string,
+    noElement: boolean = false
+  ): Promise<TemplateResult> {
+    let element: LovelaceCard | undefined;
+    let cardConfig: LovelaceCardConfig = { type };
 
-    const elClass = await getCardElementClass(type);
-    let config: LovelaceCardConfig = { type };
+    if (this.hass && this.lovelace) {
+      cardConfig = await getCardStubConfig(this.hass, this.lovelace, type);
```

Comment:
We might want to compute the used and unused entities just once here and pass them on

Reply:
I was thinking that. I wasn't sure if there was a time where we would want to compute this per card

Reply:
You make it an optional parameter, if it isn't there calculate it.